plugins/org.jboss.tools.openshift.cdk.server/src/org/jboss/tools/openshift/cdk/server/core/internal/listeners/CDKDockerUtility.java

```diff
@@ -0,0 +1,83 @@
+/******************************************************************************* 
+ * Copyright (c) 2016 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/ 
+package org.jboss.tools.openshift.cdk.server.core.internal.listeners;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.eclipse.linuxtools.docker.core.DockerConnectionManager;
+import org.eclipse.linuxtools.docker.core.DockerException;
+import org.eclipse.linuxtools.docker.core.IDockerConnection;
+import org.eclipse.linuxtools.internal.docker.core.DockerConnection;
+import org.eclipse.linuxtools.internal.docker.core.DockerConnection.Builder;
+import org.eclipse.wst.server.core.IServer;
+
+public class CDKDockerUtility {
+
+	public String getNextName(IServer server, DockerConnectionManager mgr) {
+		// Cache all names
+		IDockerConnection[] cons = mgr.getConnections();
+		List<String> names = Arrays.stream(cons).map(IDockerConnection::getName).collect(Collectors.toList());
+
+		// Find a name that doesnt match existing connection
+		final String nameBase = server.getName();
+		String name = nameBase;
+		int count = 0;
+		boolean done = false;
+		while(!done) {
+			if( names.contains(name)) {
```

Comment:
'names' should be Set rather than list for quick search. Only if we can be sure that collection size is small that it does not matter. Maybe in this case it is small, but it is good to have a habit of preferring Set to List as soon as collection may be used for search.